src/isort/sorting.rs

```diff
@@ -69,3 +69,17 @@ pub fn cmp_import_from(import_from1: &ImportFromData, import_from2: &ImportFromD
         }
     })
 }
+
+/// Compare two `AnyImport` enums which may be `Import` or `ImportFrom` structs.
+pub fn cmp_any_import(a: &AnyImport, b: &AnyImport) -> Ordering {
+    match (&a, &b) {
+        (AnyImport::Import(x), AnyImport::Import(y)) => cmp_modules(&x.0, &y.0),
+        (AnyImport::ImportFrom(x), AnyImport::Import(y)) => {
+            natord::compare_ignore_case(x.0.module.unwrap(), y.0.name)
+        }
+        (AnyImport::Import(x), AnyImport::ImportFrom(y)) => {
+            natord::compare_ignore_case(x.0.name, y.0.module.unwrap())
+        }
+        (AnyImport::ImportFrom(x), AnyImport::ImportFrom(y)) => cmp_import_from(&x.0, &y.0),
+    }
+}
```

Comment:
Guessing that `.unwrap()` could be unsafe here, although I don't understand the case that results in `module` being `None`.

Reply:
That happens for `from . import foo` (or `..`, `...`, etc.).

Reply:
Makes sense.
I threw one of those into my test cases but I suppose that since the block in the case is only homogeneous `import from` statements we never hit these middle conditions.

Reply:
I dug into this a bit.

Given the current callers, `.unwrap()` will never return `None` because relative imports _always_ use the `from X import Y` syntax. An import like `import .my_module` is invalid.
Relative imports are sorted in their own section and aren't compared to other first-party imports.

That doesn't necessarily guarantee this code path won't be hit with a missing module given a different set of settings down the road.

What's the best way to handle this, something like?
```rs
fn cmp_option_str(a: Option<&String>, b: Option<&String>) -> Ordering {
    match (a, b) {
        (None, None) => Ordering::Equal,
        (None, Some(_)) => Ordering::Less,
        (Some(_), None) => Ordering::Greater,
        (Some(x), Some(y)) => natord::compare_ignore_case(x, y),
    }
}
```

Reply:
Yeah we should do something like what you have there, just to be safe. You can see some examples of that `None` handling in functions like `cmp_level`.